packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -47,25 +48,256 @@ def _create_file(
+
+
+def _create_class_name(base_service_class: str) -> str:
+    return base_service_class.replace("_", "") + "Client"
+
+
+def _get_selected_measurement_service_class(
+    selection: int, measurement_service_classes: List[str]
+) -> str:
+    if not (1 <= selection <= len(measurement_service_classes)):
+        raise click.ClickException(
+            f"Input {selection} is not invalid. Please try again by selecting a valid measurement from the list."
+        )
+    return measurement_service_classes[selection - 1]
+
+
+def _create_client(
+    discovery_client: DiscoveryClient,
+    channel_pool: GrpcChannelPool,
+    measurement_service_class: str,
+    module_name: str,
+    class_name: str,
+    directory_out: pathlib.Path,
+) -> None:
+    built_in_import_modules: List[str] = []
+    custom_import_modules: List[str] = []
+
+    measurement_service_stub = get_measurement_service_stub(
+        discovery_client, channel_pool, measurement_service_class
+    )
+    metadata = measurement_service_stub.GetMetadata(v2_measurement_service_pb2.GetMetadataRequest())
+    configuration_metadata = get_configuration_metadata_by_index(
+        metadata, measurement_service_class
+    )
+    output_metadata = get_output_metadata_by_index(metadata)
+
+    configuration_parameters_with_type_and_default_values, measure_api_parameters = (
+        get_configuration_parameters_with_type_and_default_values(
+            configuration_metadata, built_in_import_modules
+        )
+    )
+    output_parameters_with_type = get_output_parameters_with_type(
+        output_metadata, built_in_import_modules, custom_import_modules
+    )
+
+    _create_file(
+        template_name="measurement_plugin_client.py.mako",
+        file_name=f"{module_name}.py",
+        directory_out=directory_out,
+        class_name=class_name,
+        display_name=metadata.measurement_details.display_name,
+        configuration_metadata=configuration_metadata,
+        output_metadata=output_metadata,
+        service_class=measurement_service_class,
+        configuration_parameters_with_type_and_default_values=configuration_parameters_with_type_and_default_values,
+        measure_api_parameters=measure_api_parameters,
+        output_parameters_with_type=output_parameters_with_type,
+        built_in_import_modules=to_ordered_set(built_in_import_modules),
+        custom_import_modules=to_ordered_set(custom_import_modules),
+    )
+
+    print(
+        f"The measurement plug-in client for the service class '{measurement_service_class}' has been created successfully."
+    )
+
+
+def _create_all_clients(directory_out: Optional[str]) -> None:
+    channel_pool = GrpcChannelPool()
+    discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
+
+    directory_out_path = _resolve_output_directory(directory_out)
+    measurement_service_class, _ = get_all_registered_measurement_info(discovery_client)
+    if len(measurement_service_class) == 0:
```

Comment:
Can we please keep this as a helper method and utilize it? We are using this validation in more than one place. So, in the future, if we want to modify the error message, we would need to change it in all occurrences.